SIL.DblBundle/Usx/UsxNode.cs

```diff
@@ -1,67 +1,147 @@
 		/// </summary>
-		public string StyleTag
-		{
-			get { return m_node.Attributes.GetNamedItem("style").Value; }
-		}
+		public virtual string StyleTag => GetAttribute("style").Value;
 
 		/// <summary>
 		/// List of nodes that are children of this node.
 		/// </summary>
-		public XmlNodeList ChildNodes
+		public XmlNodeList ChildNodes => m_node.ChildNodes;
+	}
+
+	/// <summary>
+	/// A USX node that specifies a chapter start or end
+	/// </summary>
+	public sealed class UsxPara : UsxNode
+	{
+		/// <summary>
+		/// Creates a an object representing a USX paragraph based on a USX node.
+		/// </summary>
+		public UsxPara(XmlNode node) : base(node)
 		{
-			get { return m_node.ChildNodes; }
+			if (node.Name != kParaNodeName)
+				throw new ArgumentException($"Not a valid {kParaNodeName} node.",
+					nameof(node));
 		}
 	}
 
 	/// <summary>
-	/// A USX node that specifies the chapter
+	/// A USX node that specifies the start or end of a run of text marked with a character style
 	/// </summary>
-	public class UsxChapter : UsxNode
+	public sealed class UsxChar : UsxNode
 	{
 		/// <summary>
-		/// Creates a chapter XML node 
+		/// Creates a an object representing a USX paragraph based on a USX node.
 		/// </summary>
-		public UsxChapter(XmlNode paraNode)
-			: base(paraNode)
+		public UsxChar(XmlNode node) : base(node)
 		{
+			if (node.Name != kCharNodeName)
+				throw new ArgumentException($"Not a valid {kCharNodeName} node.",
+					nameof(node));
 		}
+	}
 
+	/// <summary>
+	/// A USX node that specifies a chapter start or end
+	/// </summary>
+	public sealed class UsxChapter : UsxNode
+	{
 		/// <summary>
-		/// Gets the chapter number
+		/// Creates a an object representing the start or end of a chapter based
+		/// on a USX node.
 		/// </summary>
-		public string ChapterNumber
+		public UsxChapter(XmlNode node) : base(node)
 		{
-			get { return Node.Attributes.GetNamedItem("number").Value; }
+			if (node.Name != kChapterNodeName)
+				throw new ArgumentException($"Not a valid {kChapterNodeName} node.",
+					nameof(node));
 		}
 
+		/// <summary>
+		/// Gets the style tag (SFM without the leading backslash) of this node.
+		/// </summary>
+		/// <remarks>chapter end nodes do not correspond to any specific marker in USFM and
+		/// therefore do not have a StyleTag defined. Rather than returning <c>null</c>, we could
+		/// return "c" or a fictitious "c*" or even throw an exception (as we used to do), but
+		/// <c>null</c> seems to best represent the situation.</remarks>
+		public override string StyleTag => IsChapterStart ? base.StyleTag : null;
+
+		/// <summary>
+		/// Gets whether this node represents the start (as opposed to the end) of a chapter.
+		/// </summary>
+		/// <remarks>Starting with USX v. 3, chapter nodes can have an "eid" attribute (and no
+		/// "number" attribute) to indicate that the end of a chapter. A chapter node should
+		/// never have both and "sid" and an "eid" attribute, but to accommodate older versions
```

Comment:
```suggestion
		/// never have both an "sid" and an "eid" attribute, but to accommodate older versions
```

("and" is clearly wrong. I changed it to "an" to match what's below and that _sounds_ right, but from my limited English grammar knowledge I'd have thought that it should be "a". Why is it "an"?)